src/rnpkeys/tui.cpp

```diff
@@ -152,6 +152,28 @@ ask_rsa_bitlen(FILE *input_fp)
     return result;
 }
 
+static long
+ask_elgamal_bitlen(FILE *input_fp)
+{
+    do {
+        printf(
+          "Please provide bit length of the ElGamal key (between %d and %d):\n(default %d) > ",
+          ELGAMAL_MIN_P_BITLEN,
+          ELGAMAL_MAX_P_BITLEN,
+          DEFAULT_ELGAMAL_NUMBITS);
+        long result = DEFAULT_ELGAMAL_NUMBITS;
+        if (!rnp_secure_get_long_from_fd(input_fp, result)) {
+            continue;
+        }
+        if ((result >= ELGAMAL_MIN_P_BITLEN) && (result <= ELGAMAL_MAX_P_BITLEN)) {
+            // round up to multiple of 32
+            result = ((result + 31) / 32) * 32;
```

Comment:
More common in cryptography-related software would be to see `result = ((result + 0x1f) >> 5) << 5` :)

Reply:
I used existing similar code from `ask_dsa_bitlen(FILE *input_fp)` as a reference:
https://github.com/rnpgp/rnp/blob/master/src/rnpkeys/tui.cpp#L170

Reply:
Ok, let it be as it is then, probably some minor refactoring PR could fix these afterwards.